src/amo/reducers/collections.js

```diff
@@ -642,12 +646,77 @@ export const deleteCollection = ({
   };
 };
 
+type UpdateCollectionAddonParams = {|
+  addonId: number,
+  errorHandlerId: string,
+  notes: string,
+  page: number,
+  slug: string,
+  username: string,
+|};
+
+export type UpdateCollectionAddonAction = {|
+  type: typeof UPDATE_COLLECTION_ADDON,
+  payload: UpdateCollectionAddonParams,
+|};
+
+export const updateCollectionAddon = ({
+  addonId, errorHandlerId, notes, page, slug, username,
+}: UpdateCollectionAddonParams = {}): UpdateCollectionAddonAction => {
+  invariant(addonId, 'The addonId parameter is required');
+  invariant(errorHandlerId, 'The errorHandlerId parameter is required');
+  invariant(
+    notes !== undefined && notes !== null, 'The notes parameter is required'
+  );
+  invariant(page, 'The page parameter is required');
+  invariant(slug, 'The slug parameter is required');
+  invariant(username, 'The username parameter is required');
+
+  return {
+    type: UPDATE_COLLECTION_ADDON,
+    payload: {
+      addonId, errorHandlerId, notes, page, slug, username,
+    },
+  };
+};
+
+// The DeleteCollectionAddonNotesAction is identical to the
+// UpdateCollectionAddonAction, with the only difference being that
+// we pass '' into the notes field. Therefore we can reuse the
+// UpdateCollectionAddonParams type to describe the payload.
+export type DeleteCollectionAddonNotesAction = {|
+  type: typeof DELETE_COLLECTION_ADDON_NOTES,
+  payload: UpdateCollectionAddonParams,
+|};
+
+export const deleteCollectionAddonNotes = ({
+  addonId, errorHandlerId, page, slug, username,
+}: UpdateCollectionAddonParams = {}): DeleteCollectionAddonNotesAction => {
+  invariant(addonId, 'The addonId parameter is required');
+  invariant(errorHandlerId, 'The errorHandlerId parameter is required');
+  invariant(page, 'The page parameter is required');
+  invariant(slug, 'The slug parameter is required');
+  invariant(username, 'The username parameter is required');
+
+  // For delete we set the notes field to an empty string and call update.
+  // TODO: Use a null instead when https://github.com/mozilla/addons-server/issues/7832 is fixed.
+  return {
+    type: DELETE_COLLECTION_ADDON_NOTES,
+    payload: {
+      addonId, errorHandlerId, notes: '', page, slug, username,
```

Comment:
Would it make sense to add a test case for this action creator that returns an empty string for `notes`? (so that we notice it when we'll change to `null` once the API is fixed).

Reply:
That is being tested here: https://github.com/bobsilverberg/addons-frontend/blob/6977e2f7448ce56d8a3640be13925b5d1ae9e2de/tests/unit/amo/sagas/test_collections.js#L880. This is generally (I believe) the way we test action creators.

Do you disagree?

Reply:
Yup 👍